### PR TITLE
198 - Low LCP Scores for the pages in main

### LINKED ISSUE #198: Low LCP Scores for the pages in main
We are frequently observing Low LCP scores for the pages in main, bringing down the scores for all new PRs.
This problem should be addressed with priority.

Example: 

https://main--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-m-series-readers

https://pagespeed.web.dev/analysis/https-main--moleculardevices--hlxsites-hlx-live-products-microplate-readers-multi-mode-readers-spectramax-m-series-readers/1mziri2w81?form_factor=mobile

Screenshot 2023-04-14 at 12 39 30

### PR DESCRIPTION
**Pull request by andreituicu**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Optimisations used:
1. Manually select the `Hero` block as an LCP block. Franklin loads these ones first. For pages that don't have the Hero as LCP, additional blocks will need to be added there. https://github.com/hlxsites/moleculardevices/pull/206/files#diff-fbceb58f32b99b8a5309251f8c49ac8eda82fba58b905c600c92bce1ed2a1c8cR25

2. Remove small  JS files which were on the critical path and inline their content into files which have been already loaded by the browser. Additional JS files, means additional additional roundtrips to the Franklin Server. (e.g. `header-utils.js`, `templates/config.js`). Unless there are big chunks of code that are not needed in certain context, we shouldn't split small code chunks into different files, if we need to load that code anyway.

3. Lazy load JS or CSS files only when needed. When putting `import` at the beginning of a JS or CSS file, the browser will trigger a download of those imported resources as soon as it sees them, even if they might not be required until later. by using `await import(...js)` and `loadCSS` you ensure that those resources are loaded exactly when needed and not earlier to potentially be in the critical path. 
Examples: 

**CSS**: 
* Before: https://github.com/hlxsites/moleculardevices/pull/206/files#diff-0f9e38e9fb8d1cbfea693eb13af684c4b971aa99911fa70601d3605623685c38L1
* After: https://github.com/hlxsites/moleculardevices/pull/206/files#diff-98cf7e8e6eacd23bb6365a8f194c29bed3344eb59f1a25dd986e117cbee4b696R52-R54

**JS**:
* Before: https://github.com/hlxsites/moleculardevices/pull/206/files#diff-438d7b268576bb2c9750c809c1139a6f396a97c92296b36c6201d00be0a66fb0L3
* After: https://github.com/hlxsites/moleculardevices/pull/206/files#diff-438d7b268576bb2c9750c809c1139a6f396a97c92296b36c6201d00be0a66fb0R48

4. Defer the execution of JS which is in the critical path of LCP (e.g. make it async and don't wait for it to finish)

https://github.com/hlxsites/moleculardevices/pull/206/files#diff-438d7b268576bb2c9750c809c1139a6f396a97c92296b36c6201d00be0a66fb0R31

https://github.com/hlxsites/moleculardevices/pull/206/files#diff-438d7b268576bb2c9750c809c1139a6f396a97c92296b36c6201d00be0a66fb0R47

5. (may not be easy to replicate for other contexts): 

Context:

Franklin loads sections progressively, when all the blocks from each section has finished loading. (at least in our version of the Franklin boilerplate code)
https://github.com/hlxsites/moleculardevices/blob/main/scripts/lib-franklin.js#L324-L330
https://github.com/hlxsites/moleculardevices/blob/main/styles/styles.css#L479-L482

Prerequisites:

* The hero block is always the first one when it appears on pages
* We don't have any stylings on `sections` which prevent us from arbitrarily create new ones

Optimisation:

Move the hero block into its own section via autoblocking, so it is not delayed by other blocks.
(This optimisation can be use also manually in other contexts for one of pages/blocks, when needed, provided that it doesn't break the page or create CLS).

**NOTE**: 
* There are still some sporadic TBT spikes to ~100ms in some LHS executions, which I haven't identified where they come from, they are not related or impacting the LCP, but can bring the LHS score down (they were not from the mega menu).
* Hero blocks with vidyard videos may give a higher LCP sometimes, but unfortunately we don't control the vidyard JS code.

Fix #198

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.live
- After: https://198-fix-lcp--moleculardevices--hlxsites.hlx.live

- Before: https://main--moleculardevices--hlxsites.hlx.live/lab-notes/clone-screening/challenges-of-cell-line-development-and-emerging-technologies-to-regulate-monoclonality
- After: https://198-fix-lcp--moleculardevices--hlxsites.hlx.live/lab-notes/clone-screening/challenges-of-cell-line-development-and-emerging-technologies-to-regulate-monoclonality

- Before: https://main--moleculardevices--hlxsites.hlx.live/en/assets/app-note/br/analyze-cardiomyocyte-spheroid-contractions-with-softmax-pro-software
- After: https://198-fix-lcp--moleculardevices--hlxsites.hlx.live/en/assets/app-note/br/analyze-cardiomyocyte-spheroid-contractions-with-softmax-pro-software

- Before: https://main--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-i3x-readers
- After: https://198-fix-lcp--moleculardevices--hlxsites.hlx.live/products/microplate-readers/multi-mode-readers/spectramax-i3x-readers

- Before: https://main--moleculardevices--hlxsites.hlx.live/newsroom/news/hub-organoids-to-advance-automated-intestinal-organoid-screening-technology
- After: https://198-fix-lcp--moleculardevices--hlxsites.hlx.live/newsroom/news/hub-organoids-to-advance-automated-intestinal-organoid-screening-technology

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -222,14 +226,17 @@ export async function fetchFragment(path) {
...
+  const headerBlock = loadHeader(doc.querySelector('header'));
 
   await loadBlocks(main);
 
   const { hash } = window.location;
   const element = hash ? doc.getElementById(hash.substring(1)) : false;
   if (hash && element) element.scrollIntoView();
 
+  // const megaMenuModule = await import('../blocks/header/header-megamenu.js');
+  // megaMenuModule.default(headerBlock);
```
Comment (andreituicu) [inline scripts/scripts.js:239]: These will be uncommented before merging, but they are contributing a high TBT and it is was hard to see if we reach 100% LHS with them.

File: blocks/breadcrumbs/breadcrumbs-create.js [hunk 2]
```diff
@@ -48,6 +49,10 @@ function getName(pageIndex, path, current) {
 }
 
 export default async function createBreadcrumbs(container) {
+  const breadCrumbsCSS = new Promise((resolve) => {
```
Comment (mhaack) [inline blocks/breadcrumbs/breadcrumbs-create.js:52]: why is that needed shouldn't we leave up the CSS loading to the franklin loader? If this is only about the CSS I vote for move the breadcrumb CSS to styles.css or hero.css so this trick would not be needed.

**Review by jose-correia — APPROVED** [review]

File: blocks/hero/hero.css [hunk 3]
```diff
@@ -210,6 +208,11 @@ main .hero.blog .breadcrumbs > ol {
 /* Blog Variant - Desktop -- End */
 
 @media (max-width: 767px) {
+  /* duplicated from breadcrumbs.css for performance reasons */
+  .breadcrumbs {
```
Comment (mhaack) [inline blocks/hero/hero.css:212]: Maybe better add this to styles.css? For pages without a hero but with breadcrumb.

**Review by lydiapuric — APPROVED** [review]
LGTM

File: blocks/breadcrumbs/breadcrumbs-create.js (hunk 2 above)
Comment (andreituicu) [reply blocks/breadcrumbs/breadcrumbs-create.js:52]: @mhaack if my understanding is correct, the breadcrumb is both a standalone block (added via autobocking) for pages that don't have a hero (like news & publications) and part of the hero block, in which case we import manually the JS and CSS.

@paolomoz can confirm

Comment (paolomoz) [reply blocks/breadcrumbs/breadcrumbs-create.js:52]: confirmed

**Review by paolomoz — APPROVED** [review]

File: blocks/hero/hero.css (hunk 3 above)
Comment (andreituicu) [reply blocks/hero/hero.css:212]: Those pages get it from the breadcrumb block css: https://github.com/hlxsites/moleculardevices/blob/main/blocks/breadcrumbs/breadcrumbs.css#L37-L39

but since we lazy load the breadcrumb on pages with hero to optimise the LCP, it was creating CLS on mobile. That's why I also copied just this part it in the hero block css too.

File: blocks/breadcrumbs/breadcrumbs-create.js (hunk 2 above)
Comment (mhaack) [reply blocks/breadcrumbs/breadcrumbs-create.js:52]: ok

**Review by mhaack — APPROVED** [review]
